### PR TITLE
Lint rules for typescript

### PR DESCRIPTION
**Pull request by piotr-gawlowski**
I think we could use same repo for .tslint rules. I transformed our eslint rules to typescript - not all of them have equivalents. Typescript have more rules. Also last section  refers to Angular2 "styling" rules developed by community.

### CODE REVIEW

File: angular2.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
+module.exports = {
```
Comment (Avaq) [inline angular2.js]: Can you add a `"extends": "./index"`, as done in `./jsx.js`?

File: angular2.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    "use-input-property-decorator": true,
+    "use-output-property-decorator": true,
+    "use-host-property-decorator": true,
+    "no-input-rename": true,
+    "no-output-rename": true,
+    "use-life-cycle-interface": true,
+    "use-pipe-transform-interface": true,
+    "component-class-suffix": true,
+    "directive-class-suffix": true
+  }
```
Comment (Avaq) [inline angular2.js]: I think plenty of the rules here are duplicated from `./index`. No need to include them.

**Comment by piotr-gawlowski** [comment]
Hmm.. I think this might require separate repo in the end.
tslint fails trying to import
extends: 'eslint:recommended', which isn't a file with rules and kind of breaks entire logic of extends.

**Comment by Avaq** [comment]
I'm curious. I'll join you tomorrow to have a look at it. :)

**Comment by Avaq** [comment]
Actually never mind. I thought you would use something like https://github.com/nzakas/eslint-plugin-typescript. But apparently TypeScript has its own linter; https://github.com/palantir/tslint .. Since this repo is **eslint**-config-warp, I don't think tslint configuration belongs here. :\

**Pull request closed without merging**